engine/src/main/java/org/terasology/persistence/typeHandling/coreTypes/RuntimeDelegatingTypeHandler.java

```diff
@@ -56,7 +56,12 @@ public RuntimeDelegatingTypeHandler(TypeHandler<T> delegateHandler, TypeInfo<T>
     public PersistedData serializeNonNull(T value, PersistedDataSerializer serializer) {
         // If primitive, don't go looking for the runtime type, serialize as is
         if (typeInfo.getRawType().isPrimitive()) {
-            return delegateHandler.serialize(value, serializer);
+//            LOGGER.info("DelegateHandler: " + delegateHandler);
+            if (delegateHandler != null) {
+                return delegateHandler.serialize(value, serializer);
+            }
+
+            return null;
```

Comment:
Always use `serializer.serializeNull()`. Returning null `PersistedData` references will have unforeseen repercussions. With the new `TypeHandler` refactor (and all other new code I introduce in the Terasology codebase), I aim use `Optional` instead of nulls to make absent values more obvious.

Reply:
Oh, great! I can use that as part of the workaround before fixing the problem we commented above.